### PR TITLE
add device list label to view;progress idicator during scanning;scan …

### PR DESCRIPTION
**Pull request by dkwach**
…cancelation

### CODE REVIEW

File: lib/src/ble/ble_scanner.dart [hunk 1]
```diff
@@ -29,6 +30,10 @@ class BleScanner extends StatefulStream<BleScanState> {
       addStateToStream(state);
     }, onError: (Object e) {});
     addStateToStream(state);
+
+    Future.delayed(_scanDuration, () {
+      stopScan();
```
Comment (vovagorodok) [inline lib/src/ble/ble_scanner.dart:35]: Stop scan here doesn't call `Wakelock.disable();`. And I'm not sure if  stop after 10sec is needed

File: lib/src/ui/scaner_screen.dart [hunk 2]
```diff
@@ -57,6 +57,8 @@ class ScanerScreenState extends State<ScanerScreen> {
             builder: (context, snapshot) => Column(
               mainAxisAlignment: MainAxisAlignment.end,
               children: [
+                if (bleScanner.state.discoveredDevices.isNotEmpty)
```
Comment (vovagorodok) [inline lib/src/ui/scaner_screen.dart:60]: We can remove `if`. Text can be visible always

File: lib/src/ui/scaner_screen.dart [hunk 3]
```diff
@@ -57,6 +57,8 @@ class ScanerScreenState extends State<ScanerScreen> {
...
+                  const Text("Device list"),
```
Comment (vovagorodok) [inline lib/src/ui/scaner_screen.dart:61]: `Device list` -> `Devices:`.  fontSize can be higher, 20 I guess. And add some extra space after text `const SizedBox(height: 8),`

File: lib/src/ui/scaner_screen.dart (hunk 2 above)
Comment (dkwach) [reply lib/src/ui/scaner_screen.dart:60]: Agree

File: lib/src/ui/scaner_screen.dart (hunk 3 above)
Comment (dkwach) [reply lib/src/ui/scaner_screen.dart:61]: I will do this

File: lib/src/ble/ble_scanner.dart (hunk 1 above)
Comment (dkwach) [reply lib/src/ble/ble_scanner.dart:35]: Most of tools, which scan, have some timout

File: lib/src/ui/scaner_screen.dart [hunk 4]
```diff
@@ -84,6 +86,8 @@ class ScanerScreenState extends State<ScanerScreen> {
                         .toList(),
                   ),
                 ),
+                if (bleScanner.state.scanIsInProgress)
+                  const CircularProgressIndicator(),
```
Comment (vovagorodok) [inline lib/src/ui/scaner_screen.dart:90]: Hmm, not sure if additional progress indicator doesn't overload UI. We know that scan in progress from incactive start and active stop button

File: lib/src/ble/ble_scanner.dart (hunk 1 above)
Comment (dkwach) [reply lib/src/ble/ble_scanner.dart:35]: I will think how to resolve wakelock call

Comment (vovagorodok) [reply lib/src/ble/ble_scanner.dart:35]: Hmm, interesting what the reason

File: lib/src/ui/scaner_screen.dart (hunk 4 above)
Comment (dkwach) [reply lib/src/ui/scaner_screen.dart:90]: This indicate that something is happening and app is still alive. I would keep it. This animation should not be to complex to ui.

Comment (vovagorodok) [reply lib/src/ui/scaner_screen.dart:90]: For end user it additional element to look on

Comment (vovagorodok) [reply lib/src/ui/scaner_screen.dart:90]: Maybe `LinearProgressIndicator` at the top of the list will not overload interface to mach.

Comment (vovagorodok) [reply lib/src/ui/scaner_screen.dart:90]: Maybe three dots animation after last element/card of ListView.
Something like: https://pub.dev/packages/progress_indicators. Or even better: https://pub.dev/packages/dots_indicator

Comment (dkwach) [reply lib/src/ui/scaner_screen.dart:90]: I would keep current indicator (to not add external dependencies)

File: lib/src/ui/scaner_screen.dart (hunk 3 above)
Comment (dkwach) [reply lib/src/ui/scaner_screen.dart:61]: We can close this thread - As we discussed Device list would part of app bar

File: lib/src/ui/scaner_screen.dart (hunk 2 above)
Comment (dkwach) [reply lib/src/ui/scaner_screen.dart]: Removed together with text widget (device list is going to be appBar part (as we discussed))

File: lib/src/ble/ble_scanner.dart (hunk 1 above)
Comment (dkwach) [reply lib/src/ble/ble_scanner.dart]: fixed

**Pull request merged**